docs: add docs for managed EFS

bvtujo opened the pull request:
<!-- Provide summary of changes -->

<!-- Issue number, if available. E.g. "Fixes #31", "Addresses #42, 77" -->

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

bvtujo committed:
docs: add docs for managed EFS

bvtujo committed:
chore: add examples of managed EFS to common fields

huanjani commented on site/content/docs/developing/storage.en.md:
```diff
@@ -41,7 +41,105 @@ $ copilot storage init -n my-cluster -t Aurora -w api --engine PostgreSQL --init
+
+This manifest will result in an EFS volume being created at the environment level, with an Access Point and dedicated directory at the path `/frontend` created specifically for your service. This path and EFS volume will persist until you delete your environment. 
+
+The use of an access point for each service ensures that no two services can access each others' data unless you specifically intend for them to do so by specifying the full advanced configuration. You can read more in [Advanced Use Cases](#advanced-use-cases).
```
nit:
```suggestion
The use of an access point for each service ensures that no two services can access each other's data unless you specifically intend for them to do so by specifying the full advanced configuration. You can read more in [Advanced Use Cases](#advanced-use-cases).
```

bvtujo committed:
chore: address feedback

- replace `enabled` with `true`
- clarify the difference between EFS path and container FS path

efekarakus commented on site/content/docs/developing/storage.en.md:
```diff
@@ -41,7 +41,105 @@ $ copilot storage init -n my-cluster -t Aurora -w api --engine PostgreSQL --init
+storage:
+  volumes:
+    myVolume:
+      efs: enabled
```
```suggestion
      efs: true
```

huanjani commented on site/content/docs/include/common-svc-fields.md:
```diff
@@ -158,15 +158,30 @@ Required. Specify the location in the container where you would like your volume
 
-<span class="parent-field">volume.</span><a id="efs" href="#efs" class="field">`efs`</a> <span class="type">Map</span>  
-Specify more detailed EFS configuration.
+<span class="parent-field">volume.</span><a id="efs" href="#efs" class="field">`efs`</a> <span class="type">Bool or Map</span>  
```
nit: for consistency, can we use the full "Boolean" here and elsewhere in the `volumes` fields?

bvtujo committed:
chore: fix apostrophe usage

Co-authored-by: Janice Huang <60631893+huanjani@users.noreply.github.com>

huanjani commented on site/content/docs/include/common-svc-fields.md:
```diff
@@ -158,15 +158,30 @@ Required. Specify the location in the container where you would like your volume
-<span class="parent-field">volume.</span><a id="efs" href="#efs" class="field">`efs`</a> <span class="type">Map</span>  
-Specify more detailed EFS configuration.
+<span class="parent-field">volume.</span><a id="efs" href="#efs" class="field">`efs`</a> <span class="type">Bool or Map</span>  
+Specify more detailed EFS configuration. If specified as a Boolean, or using only the `uid` and `gid` subfields, creates a managed EFS filesystem and dedicated Access Point for this workload.
```
maybe lowercase?
```suggestion
Specify more detailed EFS configuration. If specified as a boolean, or using only the `uid` and `gid` subfields, creates a managed EFS filesystem and dedicated Access Point for this workload.
```

bvtujo committed:
chore: update capitalization to boolean

Co-authored-by: Janice Huang <60631893+huanjani@users.noreply.github.com>

bvtujo committed:
chore: use 'boolean' everywhere

huanjani reviewed (approved):
Looks great! Clear and informative, especially for a tricky and complicated feature!

mergify[bot] committed:
Merge branch 'mainline' into managed-efs-docs
